src/sagemaker/jumpstart/artifacts/image_uris.py

```diff
@@ -118,10 +119,39 @@ def _retrieve_image_uri(
     )
 
     if image_scope == JumpStartScriptScope.INFERENCE:
+        hosting_instance_type_variants = model_specs.hosting_instance_type_variants
+        if hosting_instance_type_variants:
+            image_uri = hosting_instance_type_variants.get_image_uri(
+                instance_type=instance_type, region=region
+            )
+            if image_uri is not None:
+                return image_uri
         ecr_specs = model_specs.hosting_ecr_specs
+        if ecr_specs is None:
+            raise ValueError("No value for hosting ECR specs found!")
+        JUMPSTART_LOGGER.warning(
+            "Using fallback hosting ECR specs for retrieving image uri "
+            "for JumpStart model '%s': '%s'",
+            model_id,
+            str(ecr_specs),
+        )
```

Comment:
the `scope` is called `inference`, but your error message refers to `hosting`. Within the team, we used these words interchangeably but mightn't confuse customers?

Please also spell out `specifications` in the messages.

Please also check if you should capitalize `uri`.